modules/internal/ChapelDistribution.chpl

```diff
@@ -394,5 +582,63 @@ module ChapelDistribution {
     proc dsiSupportsBulkTransferInterface() param return false;
     proc doiCanBulkTransferStride() param return false;
   }
-  
+
+  class BaseSparseArr: BaseArr {
+    type eltType;
+    param rank : int;
+    type idxType;
+
+    var dom; /* : DefaultSparseDom(?); */
+    /*var dataDom = {1..0};*/
+    var data: [dom.nnzDom] eltType;
```

Comment:
Hi, OK, just be sure to leave a note that we'd like a BaseSparseArrImpl so that BaseSparseArr can represent an interface for which subclasses can make different choices. You can just say in the comment that it wasn't clear how to make initializing a parent class's domain and array fields work right.